### PR TITLE
Throw an error when creating a list with the reserved name 'Upload'

### LINKED ISSUE #3214: Naming a list "Upload" results in `Error: There can be only one type named "Upload"`
# Bug report

## Describe the bug

When I name a list "Upload" and run `npm run dev`, I see the error:

```
Error: There can be only one type named "Upload".                              
    at assertValidSDL (/home/mwcz/projects/keystone-test/node_modules/graphql/validation/validate.js:108:11)
    at Object.buildASTSchema (/home/mwcz/projects/keystone-test/node_modules/graphql/utilities/buildASTSchema.js:71:34)
    at Object.buildSchemaFromTypeDefinitions (/home/mwcz/projects/keystone-test/node_modules/graphql-tools/dist/generate/buildSchemaFromTypeDefinitions.
js:25:28)                              
    at Object.makeExecutableSchema (/home/mwcz/projects/keystone-test/node_modules/graphql-tools/dist/makeExecutableSchema.js:26:29)
    at ApolloServer.initSchema (/home/mwcz/projects/keystone-test/node_modules/apollo-server-core/dist/ApolloServer.js:272:49)
    at new ApolloServerBase (/home/mwcz/projects/keystone-test/node_modules/apollo-server-core/dist/ApolloServer.js:194:30)
    at new ApolloServer (/home/mwcz/projects/keystone-test/node_modules/apollo-server-express/dist/ApolloServer.js:60:9)
    at createApolloServer (/home/mwcz/projects/keystone-test/node_modules/@keystonejs/app-graphql/lib/apolloServer.js:144:18)
    at GraphQLApp.prepareMiddleware (/home/mwcz/projects/keystone-test/node_modules/@keystonejs/app-graphql/index.js:23:20)
    at /home/mwcz/projects/keystone-test/node_modules/@keystonejs/keystone/lib/Keystone/index.js:697:17
    at Array.map (<anonymous>)
    at Keystone._prepareMiddlewares (/home/mwcz/projects/keystone-test/node_modules/@keystonejs/keystone/lib/Keystone/index.js:696:12)
    at Keystone.prepare (/home/mwcz/projects/keystone-test/node_modules/@keystonejs/keystone/lib/Keystone/index.js:714:36)
    at executeDefaultServer (/home/mwcz/projects/keystone-test/node_modules/@keystonejs/keystone/bin/utils.js:149:42)
```

## To Reproduce

1. `npm init keystone-app my-app`
2. Name: `my-app`
3. Choose "Starter (Users + Authentication)"
4. Mongo (test connection Yes)
5. Dependencies are installed with npm
6. cd my-app
7. Edit `index.js`, add the following createList call just below the existing createList call for Users.
```
keystone.createList("Upload", {
  fields: {
    name: {
      type: Text
    }
  }
});
```

Then run `npm run dev`.  The "there can only be one" error (see above) appears.

## Expected behaviour

Keystone server launches.

## System information

- OS: Fedora 30
- Node versions:
```
{
  '@keystonejs/example-projects-starter': '5.0.11',
  npm: '6.10.2',
  node: '12.9.1'
}
```

## Additional context

Changing the list name to anything besides "Upload" fixes the issue.

### PR DESCRIPTION
**Pull request by Vultraz**
Resolves #3214.

### CODE REVIEW

File: packages/keystone/lib/Keystone/index.js [hunk 1]
```diff
@@ -268,6 +268,13 @@ module.exports = class Keystone {
       throw new Error(`Invalid list name "${key}". List names cannot start with an underscore.`);
     }
 
+    // Apollo Server automatically adds an 'Upload' scalar type to the GQL schema. Since list output
+    // types are named after their keys, having a list name 'Upload' will clash and cause a confusing
+    // error on start.
+    if (key === 'Upload' || key === 'upload') {
+      throw new Error(`"Upload" cannot be used as a list name.`);
```
Comment (timleslie) [inline packages/keystone/lib/Keystone/index.js:275]:
```suggestion
      throw new Error(`Invalid list name "Upload": Built-in GraphQL types cannot be used as a list name.`);
```

Comment (timleslie) [reply packages/keystone/lib/Keystone/index.js:275]: For consistency with the error message on line 268 and also to explain why `Upload` cant be used. This of course raises the question of whether we should check for other built in names like `Query`, `Mutation`, `String`, etc 🤔

**Review by timleslie — CHANGES_REQUESTED** [review]

**Comment by timleslie** [comment]
Also, you'll see failing CI tests. This is because we now have tests which require secrets which are only available on the main repo, not on forks. We're trying to work out the best way to work around this, but in the mean time I'll do manual testing on your branch and force push if it's all 👍

File: packages/keystone/lib/Keystone/index.js (hunk 1 above)
Comment (Vultraz) [reply packages/keystone/lib/Keystone/index.js]: Done.

**Review by timleslie — APPROVED** [review]
Looks good 👍

**Pull request merged**